tensorflow/compiler/xla/tests/fusion_test.cc

```diff
@@ -522,6 +522,39 @@ XLA_TEST_F(FusionTest, DISABLED_ON_CPU(ReduceWindow)) {
       *ExecuteAndTransfer(std::move(hlo_module), {}));
 }
 
+XLA_TEST_F(FusionTest, SharedConstant) {
+  auto hlo_module = CreateNewModule();
```

Comment:
done.  

as it happens, the CreateFusionInstruction does the right thing at the moment (provided the list of operations is in the right order).  but nowhere it is asserted that the contents of the fusion will be the same in this case. 

I came upon it because I had the constant at a different place in the list at first, and in that case you get duplication - the CreateFusionInstruction is iterative, and can make quite a mess when partially complete.